libmisc/user_busy.c

```diff
@@ -106,6 +107,54 @@ static int user_busy_utmp (const char *name)
 #endif				/* !__linux__ */
 
 #ifdef __linux__
+#ifdef ENABLE_SUBIDS
+#define in_parentuid_range(uid) ((uid) >= parentuid && (uid) < parentuid + range)
+static int in_mapping (const char *sname, unsigned long ruid, unsigned long euid, unsigned long suid)
+{
+	/* 41: /proc/xxxxxxxxxx/task/xxxxxxxxxx/uid_map + \0 */
+	char path[41];
+	char buf[512], buf2[512];
+	ssize_t llen1, llen2;
+	FILE *file;
+
+	snprintf (path, 41, "/proc/%s/ns/user", sname);
+
+	if ((llen1 = readlink (path, buf, sizeof(buf))) == -1)
+		return 0;
+
+	if ((llen2 = readlink ("/proc/self/ns/user", buf2, sizeof(buf2))) == -1)
+		return 0;
+
+	if (llen1 == llen2 && memcmp(buf, buf2, llen1) == 0)
+		return 0; /* same namespace */
```

Comment:
This part makes sense - if the process is in the same namespace then it can't be using one of the subuids.  But,